docs/sql-migration-guide-upgrade.md

```diff
@@ -17,6 +17,8 @@ displayTitle: Spark SQL Upgrading Guide
 
   - The `ADD JAR` command previously returned a result set with the single value 0. It now returns an empty result set.
 
+  - In Spark version 2.4 and earlier, users can create map values with map type key via built-in function like `CreateMap`, `MapFromArrays`, etc. Since Spark 3.0, it's not allowed to create map values with map type key.
+
```

Comment:
I think should we also add this note?

> Note that, maps with map type key still exist, via reading from parquet files, converting from scala/java map, etc. Spark is not to completely forbid map as map key, but to avoid creating it by Spark itself.